src/encode_packed.rs

```diff
@@ -0,0 +1,301 @@
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z, AA }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z, AA, AB }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z, AA, AB, AC }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z, AA, AB, AC, AD }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z, AA, AB, AC, AD, AE }
+impl_encode_packed_for_tuple! { A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z, AA, AB, AC, AD, AE, AF }
+
+macro_rules! impl_encode_packed_for_i256 {
+    ($($t:ty,)*) => {$(
+        impl EncodePacked for $t {
+            fn packed_size(&self) -> usize {
+                32
+            }
+
+            fn encode_packed(&self, out: &mut [u8]) {
+                self.to_be_bytes().encode_packed(out)
+            }
+        }
+    )*};
+}
+
+impl_encode_packed_for_i256! {
+    I256,
+    U256,
+}
+
+impl EncodePacked for bool {
+    fn packed_size(&self) -> usize {
+        1
+    }
+
+    fn encode_packed(&self, out: &mut [u8]) {
+        out[0] = *self as u8;
+    }
+}
+
+impl EncodePacked for Address {
+    fn packed_size(&self) -> usize {
+        20
+    }
+
+    fn encode_packed(&self, out: &mut [u8]) {
+        out.copy_from_slice(&**self);
+    }
+}
+
+impl EncodePacked for Digest {
+    fn packed_size(&self) -> usize {
+        32
+    }
+
+    fn encode_packed(&self, out: &mut [u8]) {
+        out.copy_from_slice(&**self);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use hex_literal::hex;
+
+    #[test]
+    fn encode_basic_types() {
+        assert_eq!(encode_packed(&0u8), hex!("00"));
+        assert_eq!(encode_packed(&255u8), hex!("ff"));
+        assert_eq!(encode_packed(&0x1234u16), hex!("1234"));
+        assert_eq!(encode_packed(&0x01020304i32), hex!("01020304"));
+        assert_eq!(encode_packed(&-1i32), hex!("ffffffff"));
+        let encoded = encode_packed(&1usize);
+        // Generate hex literal of length based on usize
+        let usize_result = if std::mem::size_of::<usize>() == 4 {
+            hex!("00000001").to_vec()
+        } else {
+            hex!("0000000000000001").to_vec()
+        };
```

Comment:
One other small nit: this assumes that `size_of::<usize>()` is either `4` or `8`. While this is currently true for supported `rustc` targets, it is a shaky assumption IMO that doesn't always hold, and can be changed to not have any assumptions on the "allowed" sizes:


```suggestion
        let usize_result = {
            let mut bytes = [0_u8; std::mem::size_of::<usize>()];
            *bytes.last_mut().unwrap() = 1;
            bytes
        };
```